proto/osmosis/incentives/group_gauge.proto

```diff
@@ -0,0 +1,58 @@
+syntax = "proto3";
+package osmosis.incentives;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
+import "cosmos/base/v1beta1/coin.proto";
+import "osmosis/lockup/lock.proto";
+
+option go_package = "github.com/osmosis-labs/osmosis/v19/x/incentives/types";
+
+// SplittingPolicy determines the way we want to split incentives in groupGauges
+enum SplittingPolicy {
+  option (gogoproto.goproto_enum_prefix) = false;
+
+  Volume = 0;
+  Evenly = 1;
+}
+
+// Note that while both InternalGaugeInfo and InternalGaugeRecord could
+// technically be replaced by DistrInfo and DistrRecord from the pool-incentives
+// module, we create separate types here to keep our abstractions clean and
+// readable (pool-incentives distribution abstractions are used in a very
+// specific way that does not directly relate to gauge logic). This also helps
+// us sidestep a refactor to avoid an import cycle.
+message InternalGaugeInfo {
+  string total_weight = 1 [
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.moretags) = "yaml:\"total_weight\"",
+    (gogoproto.nullable) = false
+  ];
```

Comment:
Total/cumulative weight serves as a snapshot of the volume tracker/accumulator for the pool being distributed to. If we did not track this information, we would not be able to update weights to reflect the growth since last epoch, as we would have no way of knowing what the value was at last epoch.

While we could technically capture this value in the volume tracker itself (e.g. with a `valueAtLastEpoch` field on the pool's volume tracker), this would require linear iteration over all pools every epoch to update them to avoid pools falling out of sync when no group gauges are distributing to them.

Reply:
Just realized I explained why cumulative weights are necessary but not total (will capture the discourse above in proto comments as per your request in another comment).

We track total weight to avoid needing to do a full linear iteration over all weights before calculating pro rata amounts to distribute. It might be possible to avoid storing this in state e.g. by returning the total from the sync function, but this design pattern:
1. Is in line with our existing abstractions (e.g. distr records) 
2. Might be useful in the future for different splitting policies, where if the optimization above for some reason does not work then we would need a state migration
3. (minor point) Allows us to run invariant checks on different states to ensure total is being tracked/summed correctly

Of course feel free to challenge these if you think this is unnecessary state – would be happy to remove in a future PR!

Reply:
Thanks for the breakdown. I agree with the decisions.

Tracking this thread as TODO to transfer context over to [TDD](https://docs.google.com/document/d/1gF50VwO3Qxdz9f-PFCLUbwUCBFbVfUspoXdBni9lzWM/edit). Will think about trade-offs more while transferring